components/datetime/src/provider/calendar/symbols.rs

```diff
@@ -134,18 +87,183 @@ macro_rules! symbols {
                 #[cfg_attr(feature = "provider_serde", serde(borrow))]
                 pub stand_alone: Option<StandAloneWidthsV1<'data>>,
             }
-        };
-    }
+        }
+    };
+}
 
-symbols!(months, [Cow<'data, str>; 12]);
+symbols!(
+    months,
+    pub struct SymbolsV1<'data>(
+        #[cfg_attr(
+            feature = "provider_serde",
+            serde(borrow, with = "serde_with::As::<[serde_with::BorrowCow; 12]>")
+        )]
+        pub [Cow<'data, str>; 12],
+    );
+);
 
-symbols!(weekdays, [Cow<'data, str>; 7]);
+symbols!(
+    weekdays,
+    pub struct SymbolsV1<'data>(
+        #[cfg_attr(
+            feature = "provider_serde",
+            serde(borrow, with = "serde_with::As::<[serde_with::BorrowCow; 7]>")
+        )]
+        pub [Cow<'data, str>; 7],
+    );
+);
 
 symbols!(
-    day_periods {
-        am: Cow<'data, str>,
-        pm: Cow<'data, str>,
-        noon: Option<Cow<'data, str>>,
-        midnight: Option<Cow<'data, str>>,
+    day_periods,
+    pub struct SymbolsV1<'data> {
+        #[cfg_attr(feature = "provider_serde", serde(borrow))]
+        pub am: Cow<'data, str>,
+        #[cfg_attr(feature = "provider_serde", serde(borrow))]
+        pub pm: Cow<'data, str>,
+        #[cfg_attr(
+            feature = "provider_serde",
+            serde(borrow, with = "serde_with::As::<Option<serde_with::BorrowCow>>")
+        )]
+        pub noon: Option<Cow<'data, str>>,
+        #[cfg_attr(
+            feature = "provider_serde",
+            serde(borrow, with = "serde_with::As::<Option<serde_with::BorrowCow>>")
+        )]
+        pub midnight: Option<Cow<'data, str>>,
     }
 );
+
+#[cfg(all(test, feature = "provider_serde"))]
+mod test {
```

Comment:
thought: a cool test to perform (not in this PR) would be to use rust's custom per-collection allocator support to write a wrapping allocator that keeps track of memory allocated, and then construct a postcard based fs/static data provider and deserialize absolutely everything, printing out how much heap memory each component takes.